HAR-3941 Elykohtainen laskutusyhteenveto

jarnovayrynen opened the pull request.

jarnovayrynen committed:
Laske elykohtainen lask.yht.veto

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Vähennä lokitusta

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Lisää laskutusyhteenvetoon raporttitesti

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Ehosta lask.yht.vedon datavirheiden tunnistusta

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Paranna lask.yht.vetoraportin kommentteja

jarnovayrynen committed:
Korjaa lask.yht.veto cachen invalidointia indeksien osalta

Jos peruslukuun vaikuttavia arvoja muutetaan, invalidoidaan kakku.

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Huomioi kaikki kentät laskennoissa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Korjaa välilyönnit

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Poista JVM-optio lokittamisesta

jarnovayrynen committed:
Lisää testi elyn lask.yht.veto -raportille

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Lisää testejä elyn lask.yht.veto -raportille

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Tunnista, millä urakoista ind.lask. epäonn. ind.arvon puutteeseen

jarnovayrynen committed:
Korjaa suolasakon puuttumisen hallinta lask.yht.vedossa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3941

jarnovayrynen committed:
Suojele molemmat parametrit nilliltä summauksessa

jarnovayrynen committed:
Lisää ely-lask.yht.vetoon testi jossa indeksi puuttuu

jarnovayrynen committed:
Pidä kirjaa miten indeksipuute sotkee LYV-laskenta

Paranna Laskutusyhteenveto-raportin virhekäsittelyä:
Kerää talteen urakat, joiden laskentaan indeksipuute vaikutti.
Kerää talteen kentät, joiden laskentaan indeksipuute vaikutti.
Käytä eksplisiittistä tietoa virheistä raportin näyttämisessä entisen
nil-tiedon hyödyntämisen sijaan. Yhden urakan tapauksessa nil tarkoitti,
että indeksejä puuttui. Usean urakan tapauksessa laskemme ahneesti yhteen
eri urakoiden summia korvaten nillin nollalla. Tämän vuoksi tarviaan eril-
linen kirjanpito indeksipuutteista.

jarnovayrynen committed:
Ehosta hankalan asian dokumentointia LYV-koodissa

jarnovayrynen committed:
Korjaa urakat-joiden-laskennan-indeksipuute-sotki laskenta

tatut committed:
Merge branch 'develop' into HAR-3941

tatut commented on src/clj/harja/palvelin/raportointi/raportit/laskutusyhteenveto.clj:
```diff
@@ -202,61 +246,107 @@
+        ;; Talteen lähtötiedot, josta tutkitaan mahdolliset laskentaa sotkevat puutteet. Mahdollisia puutteita:
+        ;; lämpötilatietoja puuttuu (suolasakon laskenta), indeksitietoja puuttuu (perusluku tai raportin aikaisia pistelukuja)
+        urakoiden-lahtotiedot (into
+                                (sorted-map)
```
Miksi sorted map? Miksei vaan vektori mäppejä?

jarnovayrynen replied:
Joo ei varmaan oo välttämätön. Ajatus oli säilyttää järjestystä mutta senhän olisi vektorikin tehnyt. Näetkö että on syytä muuttaa?

tatut commented on src/clj/harja/palvelin/raportointi/raportit/laskutusyhteenveto.clj:
```diff
@@ -187,10 +190,51 @@
   (log/debug "LASKUTUSYHTEENVETO PARAMETRIT: " (pr-str parametrit))
-  (let [{urakan-nimi :nimi indeksi :indeksi} (first (urakat-q/hae-urakka db urakka-id))
-        indeksi-kaytossa? (some? indeksi)
+  (let [konteksti (cond urakka-id :urakka
```
Tämä let lohko on jo ihan liian pitkä, voisi yrittää pilkkoa monimutkaisempia päättelyitä omiin apurifunktioihin.

jarnovayrynen committed:
Siirrä Lask.yht.vedon logiikkaa funktioihin

Siirrä Lask.yht.vedon logiikkaa let-lohkosta funktioihin. Halutaan
 parantaa let-lohkon luettavuutta.

jarnovayrynen committed:
Merge remote-tracking branch 'origin/HAR-3941' into HAR-3941

jarnovayrynen committed:
Uudelleenryhmittele Lask.yht.vedon let-lohkoa

jarnovayrynen committed:
Puolustaudu jos ELY:n LYV-raportissa ei aktiivisia urakoita

jarnovayrynen committed:
Lisää lokitusta satunnaisesti feilaavaan testiin

tatut committed:
Rename V1_595__.sql to V1_596__.sql

tatut committed:
Rename V1_596__.sql to V1_597__.sql

tatut committed:
Merge branch 'develop' into HAR-3941

jarnovayrynen committed:
Merge branch 'develop' into HAR-3941
